labrad/concurrent.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import
+
+import warnings
+
+from concurrent.futures import Future, CancelledError
+
+
+def map_future(src, func, *args, **kw):
```

Comment:
I like map_future, but chain_future might make more sense to people not used to maps.

However, I strongly think that map_future should return a MappedFuture or chain_future should return a ChainedFuture.